metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -164,6 +170,100 @@ def build_plan(
+        # Time range is constrained when joined by the base data source
+        conversion_measure_recipe = self._find_measure_recipe(
+            measure_spec_properties=conversion_measure_properties,
+            time_range_constraint=None,
+            linkable_specs=queried_linkable_specs.as_tuple,
+        )
+
+        base_measure_node = base_measure_recipe.measure_node
+        conversion_measure_node = conversion_measure_recipe.measure_node
+
+        primary_key_specs: Tuple[InstanceSpec, ...] = tuple(
+            LinklessIdentifierSpec.from_reference(ref)
+            for ref in self._data_source_semantics.get_primary_key_identifiers_for_data_source(
+                conversion_measure_properties.data_source_reference
+            )
+        )
+        if len(primary_key_specs) == 0:
+            conversion_measure_node = AddGeneratedUuidColumnNode(parent_node=conversion_measure_node)
+            primary_key_specs = (MetadataSpec.from_name(MetricFlowReservedKeywords.MF_INTERNAL_UUID.value),)
+
+        base_time_dimension_reference = self._data_source_semantics.get_agg_time_dimension_for_measure(
+            base_measure_spec.measure_spec.as_reference
+        )
+        conversion_time_dimension_reference = self._data_source_semantics.get_agg_time_dimension_for_measure(
+            conversion_measure_spec.measure_spec.as_reference
+        )
+        join_conversion_node = JoinConversionEventsNode[SqlDataSetT](
+            base_node=base_measure_node,
+            base_time_dimension_spec=TimeDimensionSpec.from_reference(base_time_dimension_reference),
+            conversion_node=conversion_measure_node,
+            conversion_measure_spec=conversion_measure_spec.measure_spec,
+            conversion_time_dimension_spec=TimeDimensionSpec.from_reference(conversion_time_dimension_reference),
+            conversion_primary_key_specs=primary_key_specs,
+            entity_spec=entity_spec,
+            window=window,
+        )
+        conversion_measure_recipe = MeasureRecipe(
+            measure_node=join_conversion_node,
+            required_local_linkable_specs=base_measure_recipe.required_local_linkable_specs,
+            join_linkable_instances_recipes=base_measure_recipe.join_linkable_instances_recipes,
+        )
+        conversion_aggregated_measures_node = self.build_aggregated_measures(
+            metric_input_measure_specs=(conversion_measure_spec,),
+            queried_linkable_specs=queried_linkable_specs,
+            where_constraint=where_constraint,
+            time_range_constraint=time_range_constraint,
+            cumulative=False,
+            cumulative_window=None,
+            cumulative_grain_to_date=None,
+            measure_recipe=conversion_measure_recipe,
+        )
+
+        # Combine the aggregated opportunities and conversion values
+        return JoinAggregatedMeasuresByGroupByColumnsNode(
```

Comment:
Oh interesting. And that just.... works? That's pretty nice.

Reply:
<img src="https://media0.giphy.com/media/e9naycep2pz7OHQ4n0/giphy.gif"/>